src/System.Private.ServiceModel/tests/Common/Infrastructure/src/BridgeClient.cs

```diff
@@ -66,6 +70,71 @@ public static string GetResourceAddress(string resourceName)
             return resourceAddress;
         }
 
+        // Issues a GET request to the Bridge base address to determine
+        // whether it is running.  This method also sets state to indicate
+        // whether the bridge is running or unavailable.
+        private static void PingBridge()
+        {
+            using (HttpClient httpClient = new HttpClient())
+            {
+                try
+                {
+                    var response = httpClient.GetAsync(BridgeBaseAddress).GetAwaiter().GetResult();
+                    if (!response.IsSuccessStatusCode)
+                    {
+                        string reason = String.Format("{0}Bridge returned unexpected status code='{1}', reason='{2}'",
+                                                    Environment.NewLine, response.StatusCode, response.ReasonPhrase);
+                        if (response.Content != null)
+                        {
+                            string contentAsString = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                            if (contentAsString.Length > 1000)
+                            {
+                                contentAsString = contentAsString.Substring(0, 999) + "...";
+                            }
+                            reason = String.Format("{0}, content:{1}{2}",
+                                                    reason, Environment.NewLine, contentAsString);
+                        }
+                        throw new Exception(reason);
+                    }
+                    _BridgeStatus = BridgeState.Started;
+                }
+                catch (Exception exc)
+                {
+                    _BridgeStatus = BridgeState.Faulted;
+                    throw new Exception("Bridge is not running", exc);
+                }
+            }
+        }
+
+        // Returns true if the Bridge will respond or has responded to localhost.
+        // This is used to determine whether local file paths are acceptable for the BridgeResourceFolder.
+        private static bool DoesBridgeRespondToLocalHost()
+        {
```

Comment:
I've had the build scripts launch the bridge locally even though I specified a remote host. The tests still ran against the remote host but I had a local instance still running. This code would return true in that situation.